redhat-appstudio/service-provider-integration-operator
# Refactor contoller to be more idempotent

Signed-off-by: Max Shaposhnyk <mshaposh@redhat.com>

### What does this PR do?
This PR changes the way of `SPIFileContentRequest` reconcillation in the more status-independent way.
It shifts the linking between File request and binding from the status field in the file request (updates to which may be not reflected/seen in the reconcillations that was scheduled on some intermediate resource versions of the request CR, and cause to re-wtiting status concurrently) to the specific label in the binding. 

That also having a positive effect in more precise bindings watching, avoiding processing the events from bindings which are unrelated to the file requests.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-290

### How to test this PR?

The only difference with current main is that duplicate bidndings for the same FileRequest should never appear in tests nor in live cluster anymore. (reviewers, please run `make itest focus=SPIFileContentRequest`  several times)
